client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { OnboardSelect } from '@automattic/data-stores';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { generatePath, useMatch, useNavigate } from 'react-router';
+import { useSearchParams } from 'react-router-dom';
+import { ONBOARD_STORE, STEPPER_INTERNAL_STORE } from 'calypso/landing/stepper/stores';
+import { Navigate, StepperStep } from '../../types';
+
+const useOnboardingIntent = () => {
+	const intent = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getIntent(),
+		[]
+	);
+	return intent;
+};
+
+const addQueryParams = ( uri: string, params?: URLSearchParams | null ) => {
+	if ( params ) {
+		return uri + '?' + params.toString();
+	}
+	return uri;
+};
+
+interface FlowNavigation {
+	navigate: Navigate< StepperStep[] >;
+
+	params: {
+		flow: string | null;
+		step: string | null;
+		lang: string | null;
+	};
+	search: URLSearchParams;
+}
+//TODO: Check why we need the stepPaths because there is no reference to it in the codebase.
+// If we don't need it, we might call this hook directly from the flow files, instead of drilling the navigate function, reducing the complexity and rerenders.
+/**
+ *  Hook to manage the navigation between steps in the flow
+ * @param stepPaths
+ * @returns
+ */
+
+export const useFlowNavigation = ( stepPaths: string[] ): FlowNavigation => {
```

Comment:
@daledupreez  @omarjackman  Do you know the reason we need to store the stepPaths? 
I couldn't find references on the codebase.

Reply:
The `stepPaths` can go. It seems to have started since [the very first commit to Stepper.](https://github.com/Automattic/wp-calypso/blob/b57297fac00a56cf103f3235e33f6636ddca1f49/client/landing/stepper/declarative-flow/internals/index.tsx#L22) And it most likely was just a mistake.

Reply:
Thanks! Removed here [aef0abf](https://github.com/Automattic/wp-calypso/pull/91935/commits/aef0abfe8e97a0de922621a079f1374562d3ce23)